stream_consumer.go

```diff
@@ -147,12 +218,12 @@ func (se *StreamEndpoint) ConsumeStream(streamName string, opts ...ConsumerConfi
 				Log.Debug("Stream connected", zap.String("streamName", streamName))
 
 				// at this point, the GRPC connection is established with the server
-				for {
+				for !c.stopped {
```

Comment:
you should use c.IsStopped() here, or use the c.m.RLock()
But with locks, it might be slow

Reply:
yes, that's what I intended to do